File: 백준/Silver/1904. 01타일/01타일.java
import java.util.Scanner;
import java.math.BigInteger;
public class Main{
    static int n;
    static BigInteger[] fibo;

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        fibo = new BigInteger[n+2];
        fibo[0]=new BigInteger("0");
        fibo[1]=new BigInteger("1");
        BigInteger modValue = new BigInteger("15746");
        for(int i=2;i<=n+1;i++){
            fibo[i]=((fibo[i-1]).add(fibo[i-2])).mod(modValue);
        }

        System.out.println(fibo[n+1]);
    }
}

File: 백준/Silver/1904. 01타일/01타일.py
N = int(input())

fibo = list()
fibo.append(0)
fibo.append(1)

mod = 15746
for i in range(2,N+2):
    base = fibo[i-1]+fibo[i-2]
    fibo.append(base%mod)

print(fibo[N+1])